### PR TITLE
clientv3: only use return closeErr when donec is closed

### LINKED ISSUE #5800: clientv3: race in watch
```
WARNING: DATA RACE
Read by goroutine 116:
  github.com/coreos/etcd/clientv3.(*watchGrpcStream).serveStream()
      /home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/clientv3/watch.go:578 +0x371

Previous write by goroutine 108:
  github.com/coreos/etcd/clientv3.(*watchGrpcStream).run.func1()
      /home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/clientv3/watch.go:364 +0x69
  github.com/coreos/etcd/clientv3.(*watchGrpcStream).run()
      /home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/clientv3/watch.go:441 +0xf37

Goroutine 116 (running) created at:
  github.com/coreos/etcd/clientv3.(*watchGrpcStream).addStream()
      /home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/clientv3/watch.go:345 +0x4e1
  github.com/coreos/etcd/clientv3.(*watchGrpcStream).run()
      /home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/clientv3/watch.go:398 +0x845

Goroutine 108 (finished) created at:
  github.com/coreos/etcd/clientv3.(*watcher).newWatcherGrpcStream()
      /home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/clientv3/watch.go:197 +0x5e3
  github.com/coreos/etcd/clientv3.(*watcher).Watch()
      /home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/clientv3/watch.go:229 +0x555
  github.com/coreos/etcd/clientv3/naming.(*GRPCResolver).Resolve()
      github.com/coreos/etcd/clientv3/naming/_test/_obj_test/grpc.go:86 +0x1e4
  github.com/coreos/etcd/clientv3/naming.TestGRPCResolver()
      /home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/clientv3/naming/grpc_test.go:37 +0x179
  testing.tRunner()
      /usr/local/golang/1.6.2/go/src/testing/testing.go:473 +0xdc
==================
```

### PR DESCRIPTION
**Pull request by heyitsanthony**
Fixes #5800

### CODE REVIEW

File: clientv3/watch.go [hunk 1]
```diff
@@ -505,6 +505,7 @@ func (w *watchGrpcStream) serveWatchClient(wc pb.Watch_WatchClient) {
 
 // serveStream forwards watch responses from run() to the subscriber
 func (w *watchGrpcStream) serveStream(ws *watcherStream) {
+	var closeError error
```
Comment (gyuho) [inline clientv3/watch.go]: `closeErr`?

**Comment by gyuho** [comment]
lgtm.

**Comment by xiang90** [comment]
lgtm

**Comment by xiang90** [comment]
@gyuho let's get this back-ported

**Comment by gyuho** [comment]
Ok will backport soon.

**Comment by gyuho** [comment]
backported to release-3.0

**Pull request merged**